diff -p -up bmpx-0.40.14/widgets/popup.cc~ bmpx-0.40.14/widgets/popup.cc
--- bmpx-0.40.14/widgets/popup.cc~	2008-03-05 22:31:12.000000000 +0100
+++ bmpx-0.40.14/widgets/popup.cc	2009-01-23 11:26:30.000000000 +0100
@@ -308,7 +308,7 @@ namespace Bmp
 
 		if (m_location == ARROW_TOP)
 		{
-			Cairo::Matrix matrix = { 1, 0, 0, -1, 0, h };
+		        Cairo::Matrix matrix = Cairo::Matrix( 1, 0, 0, -1, 0, h);
 			cr->set_matrix (matrix);
 		}
 
